File: weathertemp_monit.cpp
#include <Arduino.h>
#include <Wire.h>
#include <Adafruit_Sensor.h>
#include <Adafruit_BME280.h>
#include <WiFiClientSecure.h>
#include <ESP8266WiFi.h>
#include <ESP8266HTTPClient.h>
#include <Arduino_JSON.h>
#include <GSM.h>

#define BME_SDA 4
#define BME_SCL 5
#define BME_ADDRESS 0x76
#define GSM_RX 2
#define GSM_TX 3

const char* ssid = "IhrWiFiSSID";
const char* password = "IhrWiFiPasswort";
const char* serverURL = "https://example.com/api";
const char* gsmAPN = "IhrGSM-APN";
const char* gsmUser = "IhrGSM-Benutzername";
const char* gsmPass = "IhrGSM-Passwort";
const char* phoneNumber = "+1234567890";

Adafruit_BME280 bme(BME_ADDRESS);
GSM gsmAccess;

void setup() {
  Serial.begin(9600);
  Wire.begin(BME_SDA, BME_SCL);
  bme.begin(0x76);

  WiFi.begin(ssid, password);
  while (WiFi.status() != WL_CONNECTED) {
    delay(1000);
    Serial.println("Verbindung zum Wi-Fi-Netzwerk...");
  }

  gsmAccess.begin(gsmAPN, gsmUser, gsmPass);
  gsmAccess.wakeUp();
}

void loop() {
  delay(60000); // Messwerte alle 60 Sekunden senden

  if (WiFi.status() == WL_CONNECTED) {
    float temperature = bme.readTemperature();
    float humidity = bme.readHumidity();
    float pressure = bme.readPressure() / 100.0F;

    if (temperature > 30.0 || humidity > 70.0 || pressure < 900.0) {
      sendAlertSMS("Umweltbedingungen außerhalb des Normalbereichs - Temperatur, Luftfeuchtigkeit oder Druck.");
      sendAlertEmail("Umweltbedingungen außerhalb des Normalbereichs - Temperatur, Luftfeuchtigkeit oder Druck.");
    }
  }
}

void sendAlertSMS(String message) {
  GSM_SMS sms;
  if (sms.beginSMS(phoneNumber)) {
    sms.print(message);
    sms.endSMS();
  }
}

void sendAlertEmail(String message) {
  WiFiClientSecure client;
  if (client.connect(serverURL, 443)) {
    String payload = "message=" + message;
    HTTPClient http;
    http.begin(client, serverURL);
    http.addHeader("Content-Type", "application/x-www-form-urlencoded");
    http.POST(payload);
    http.end();
  }
}

File: weathertemp_monit.py
import time
from Adafruit_BME280 import BME280
import smtplib
from email.mime.text import MIMEText
import json
from twilio.rest import Client

# BME280 Sensor-Initialisierung
bme = BME280()

# Wi-Fi-Konfiguration
WIFI_SSID = 'Ihr_WiFi_SSID'
WIFI_PASSWORD = 'Ihr_WiFi_Passwort'

# SMTP-Konfiguration für E-Mail
SMTP_SERVER = 'smtp.example.com'
SMTP_PORT = 587
SMTP_USERNAME = 'Ihre_Email@example.com'
SMTP_PASSWORD = 'Ihr_Email_Passwort'
SMTP_TO = 'ziel@email.com'

# Twilio-Konfiguration für SMS
TWILIO_SID = 'Ihr_Twilio_SID'
TWILIO_AUTH_TOKEN = 'Ihr_Twilio_Authentifizierungstoken'
TWILIO_PHONE_NUMBER = 'Ihre_Twilio_Rufnummer'
DEST_PHONE_NUMBER = 'Ziel_Rufnummer'

def send_email(subject, message):
    msg = MIMEText(message)
    msg['From'] = SMTP_USERNAME
    msg['To'] = SMTP_TO
    msg['Subject'] = subject

    server = smtplib.SMTP(SMTP_SERVER, SMTP_PORT)
    server.starttls()
    server.login(SMTP_USERNAME, SMTP_PASSWORD)
    server.sendmail(SMTP_USERNAME, SMTP_TO, msg.as_string())
    server.quit()

def send_sms(message):
    client = Client(TWILIO_SID, TWILIO_AUTH_TOKEN)
    client.messages.create(
        to=DEST_PHONE_NUMBER,
        from_=TWILIO_PHONE_NUMBER,
        body=message
    )

while True:
    temperature = bme.read_temperature()
    humidity = bme.read_humidity()
    pressure = bme.read_pressure() / 100.0

    if temperature > 30.0 or humidity > 70.0 or pressure < 900.0:
        alert_message = "Umweltbedingungen außerhalb des Normalbereichs - Temperatur, Luftfeuchtigkeit oder Druck."
        send_email("Alarm - Umweltbedingungen", alert_message)
        send_sms(alert_message)

    time.sleep(60)  # Überprüfen Sie alle 60 Sekunden
